text/0014-ai-prompt-api.md

```diff
@@ -0,0 +1,176 @@
+# RFC Template
+
+- Start Date: 2025-04-09
+- RFC PR: [electron/rfcs#14](https://github.com/electron/rfcs/pull/14)
+- Electron Issues:
+- Reference Implementation:
+- Status: **Proposed**
+
+# Prompt API support
+
+## Summary
+
+The Prompt API is part of [Built-in AI](https://developer.chrome.com/docs/ai/built-in)
+capabilities in Chromium based browsers. With the Prompt API developers can send
+natural language requests from the renderer for local processing.  This proposal adds
+support for this API in Electron.
+
+## Motivation
+
+Built in AI features are being implemented in Chromium based browsers.  Electron should
+also support these features.
+
+## Guide-level explanation
+
+The Prompt API can be used by Electron app developers by defining a local AI handler
+script via `Session.registerLocalAIHandlerScript` and a new module `localAIHandler`.
+The handler script will run in a [utilityProcess](https://www.electronjs.org/docs/latest/api/utility-process)
+and will allow app developers to proxy Built-in API calls to a local LLM implementation of
+their choice.
+
+For the initial implementation, the local AI handler will only support the prompt API,
+but the long term intention is that this handler would also support other Built-in AI APIs.
+
+### Session
+
+`ses.registerLocalAIHandlerScript(options)`
+* `options` Object
+  * `filePath` string - Path of the script file. Must be an absolute path.
+
+Returns `string` - The ID of the registered local AI handler script.
+
+`ses.unregisterLocalAIHandlerScript(id)`
+
+* `id` string - local AI handler script ID
+
+Unregisters the specified local AI handler script and terminates the associated
+utility process if it is running.
+
+`ses.getUtilityProcessForLocalAI(id)`
+
+* `id` string - local AI handler script ID
+
+Returns [`UtilityProcess`](utility-process.md#class-utilityprocess) | null - if the specified
+local AI handler script has an associated utility process running it will be returned; 
+otherwise it returns null.
+
+#### Event: 'local-ai-utility-started'
+
+Returns:
+
+* `event` Event
+* `id` string - local AI handler script ID
+
+Emitted when a utility process is started for the specified local AI handler script.
+
+### localAIHandler
+
+> Proxy Built-in AI APIs to a local LLM implementation
+
+Process: [utility](https://www.electronjs.org/docs/latest/api/utility-process)
+
+This module is intended to be used by a script registered to a session via
+`ses.registerLocalAIHandlerScript(options)`
+
+#### Methods
+
+The `localAIHandler` module has the following methods:
+
+##### `localAIHandler.setPromptAPIHandler(handler)`
+* `handler` Function\<[LanguageModel](https://github.com/webmachinelearning/prompt-api?tab=readme-ov-file#full-api-surface-in-web-idl)\> | null
```

Comment:
Would it be possible to make this async? Since we're just getting a `webContentsId`, it might be necessary to make an IPC call to the Main process to gather other info.

Reply:
We may eventually want to add other parameters in future, so it might be more future-proof if we pass a dictionary to the handler, rather than positional paramters